main.go

```diff
@@ -106,55 +105,79 @@ func main() {
 	}
 	if err := checkIfRequiredFilesExist(requiredFiles); err != nil {
-		log.Error(err, "could not start the operator")
-		os.Exit(1)
-	}
-
-	// The watched namespace defined by the WMCO CSV
-	namespace, err := k8sutil.GetWatchNamespace()
-	if err != nil {
-		log.Error(err, "failed to get watch namespace")
+		setupLog.Error(err, "could not start the operator")
 		os.Exit(1)
 	}
 
 	ctx := context.TODO()
 	// Become the leader before proceeding
 	err = leader.Become(ctx, "windows-machine-config-operator-lock")
 	if err != nil {
-		log.Error(err, "failed to become a leader within current namespace")
+		setupLog.Error(err, "failed to become a leader within current namespace")
 		os.Exit(1)
 	}
 
-	// Allow for the watching of cluster-wide resources with "", so that we can watch nodes,
-	// as well as resources within the `openshift-machine-api` and WMCO namespace
-	namespaces := []string{"", "openshift-machine-api", namespace}
-	// Create a new Cmd to provide shared dependencies and start components
-	mgr, err := manager.New(cfg, manager.Options{
-		NewCache:           cache.MultiNamespacedCacheBuilder(namespaces),
+	// Create a new Manager to provide shared dependencies and start components
+	// The NewCache field is not being set, as the default is a cluster wide scope, which is what we want
+	// as we need to watch Nodes.
+	// Namespaces for each controller should be excluded using predicate functions when defining watches.
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:             scheme,
 		MetricsBindAddress: fmt.Sprintf("%s:%d", metrics.Host, metrics.Port),
+		Port:               9443,
 	})
 	if err != nil {
-		log.Error(err, "failed to create a new Manager")
+		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
-	log.Info("registering Components.")
-
 	// Setup Scheme for all resources
 	if err := apis.AddToScheme(mgr.GetScheme()); err != nil {
-		log.Error(err, "failed to add all Resources to the Scheme")
+		setupLog.Error(err, "failed to add all Resources to the Scheme")
+		os.Exit(1)
+	}
+
+	// Get the watched namespace. This is originally sourced from from the OperatorGroup associated with the CSV.
+	// Because the WMCO CSV only supports the OwnNamespace InstallMode, the watch namespace will always be the namespace
+	// that WMCO is deployed in.
+	watchNamespace, err := getWatchNamespace()
+	if err != nil {
+		setupLog.Error(err, "failed to get watch namespace")
+		os.Exit(1)
+	}
+	// Check that the OperatorGroup + CSV were not deployed with a cluster scope (namespace = ""), and that the
+	// OperatorGroup does not target multiple namespaces. This should not be able to happen as both `AllNamespaces` and
+	// `MultiNamespaces` are not supported InstallModes, and enabling those would require changes to the CSV,
+	// but including this check to be defensive, as WMCO will not function properly in that case.
+	if watchNamespace == "" || strings.Contains(watchNamespace, ",") {
```

Comment:
Wouldn't a better check `watchNamespace != operatorNamespace`?

Reply:
The operator namespace is being sourced from the watch namespace, I'm not sure where else the namespace could be taken from.

Reply:
The operator namespace needn't be the namespace watched by the operator. I thought there were env variables for each.

Reply:
By us only supporting the `OwnNamespace` deployment method, the operator namespace must be the watch namespace.
The other ENV variable is  `OPERATOR_NAME` which is the name of the operator itself, `windows machine config operator`, not the namespace.